packages/next/src/build/transpile-config.ts

```diff
@@ -0,0 +1,45 @@
+import { join } from 'path'
+import { mkdir, readFile, writeFile } from 'fs/promises'
+import { transform } from './swc'
+
+type Log = {
+  error: (message: string) => void
+}
+
+const transformOptions = {
+  module: {
+    type: 'commonjs',
+  },
+  jsc: {
+    target: 'esnext',
```

Comment:
This is still wrong. Imagine this:

```ts
// next.config.ts
import cryptoRandomString from 'crypto-random-string'; // This is a pure ESM module

export default {
  async generateBuildId() {
    return cryptoRandomString({length: 10, type: 'url-safe'});
  }
}
```

And `Error [ERR_REQUIRE_ESM]` will just pop up.

Reply:
You're right, so just added cts mts, esm by default. Thank you!

Reply:
@SukkaW One thing I'm concerned about is the path difference when the compiled file goes into the `.next` dir.

It'll surely affect it in various ways, and I'm thinking of converting it as `next.compiled.config.(ts|mts|cts)` on the same level, and add it to gitignore.

What do you think?